### ISSUE TITLE
代码中和论文不一致问题

### ISSUE THREAD

**Issue by liuyebug** [jpthu17/EMCL#1]
作者你好，我下载了您的代码，尝试去复现论文的结果，但是，我在看代码的时候，发现代码中用的数据是包含audio，face，flow，ocr等特征，但是论文中并没有提到用到这些额外的信息？能帮忙指导一下么？

**Comment by jpthu17** [comment]
代码库里的是MMT+EMCL的代码（对应于论文里table2的实验），并非EMCL-Net的代码，MMT模型是用到了audio，face，flow，ocr等特征的。论文里描述的是EMCL-Net的结构，并没有用到其他多模态特征，我们打算在后续工作发布出来之后开源完整代码，对给你带来的困惑我们表示抱歉。如果你迫切想复现我们的模型，可以给我留一个邮箱地址，我会把未经过整理的完整代码发给你。

**Comment by prote376** [comment]
Hi, I want to reproduce your work, too. 

Could you please send me the code?

If you could, my e-mail address is prote376@gmail.com

Thank you in advance!

**Comment by jpthu17** [comment]
> Hi, I want to reproduce your work, too.
> 
> Could you please send me the code?
> 
> If you could, my e-mail address is [prote376@gmail.com](mailto:prote376@gmail.com)
> 
> Thank you in advance!

Thank you for your attention to our work. I have sent the code to you, please check. If you have any further questions, please feel free to contact us. We hope to help you with your work. Anyway, we will probably release our code fully in a few months.

**Comment by liuyebug** [comment]
> 代码库里的是MMT+EMCL的代码（对应于论文里table2的实验），并非EMCL-Net的代码，MMT模型是用到了audio，face，flow，ocr等特征的。论文里描述的是EMCL-Net的结构，并没有用到其他多模态特征，我们打算在后续工作发布出来之后开源完整代码，对给你带来的困惑我们表示抱歉。如果你迫切想复现我们的模型，可以给我留一个邮箱地址，我会把未经过整理的完整代码发给你。

好的，太感谢了，我的邮箱是 lyyx000@sjtu.edu.cn

**Comment by jpthu17** [comment]
> > 代码库里的是MMT+EMCL的代码（对应于论文里table2的实验），并非EMCL-Net的代码，MMT模型是用到了audio，face，flow，ocr等特征的。论文里描述的是EMCL-Net的结构，并没有用到其他多模态特征，我们打算在后续工作发布出来之后开源完整代码，对给你带来的困惑我们表示抱歉。如果你迫切想复现我们的模型，可以给我留一个邮箱地址，我会把未经过整理的完整代码发给你。
> 
> 好的，太感谢了，我的邮箱是 [lyyx000@sjtu.edu.cn](mailto:lyyx000@sjtu.edu.cn)

已发送，注意查收。祝科研顺利。

**Comment by liuyebug** [comment]
> 祝

你好，我通过您发的代码，复现了下面图中绿色框部分的结果，但是红色框部分的结果，论文里面描述的是需要结合inverted softmax，这块不是很懂，您方便发下这块的代码么？我的邮箱是 [lyyx000@sjtu.edu.cn](mailto:lyyx000@sjtu.edu.cn)
<img width="862" alt="图1" src="https://user-images.githubusercontent.com/120089873/207500672-65d0e418-9cd1-4bf7-8367-8cb02a4e6569.png">

**Comment by jpthu17** [comment]
> 你好，我通过您发的代码，复现了下面图中绿色框部分的结果，但是红色框部分的结果，论文里面描述的是需要结合inverted softmax，这块不是很懂，您方便发下这块的代码么？

Thank you for your attention to our work. I have sent the code to you, please check. The core parts are as follows:
```c
t2v_sim_matrix = t2v_sim_matrix * F.softmax(t2v_sim_matrix / args.t2v_temp, dim=0) * len(t2v_sim_matrix)
v2t_sim_matrix = v2t_sim_matrix * F.softmax(v2t_sim_matrix / args.v2t_temp, dim=1) * len(v2t_sim_matrix)
```

**Comment by OPA067** [comment]
Hi, thank you for your great work!
I want to reproduce your work, too.

Could you please send me all code?

If you could, my e-mail address is 1719472278@qq.com, thank you!
